app/blueprints/assessments/routes.py

```diff
@@ -680,6 +683,83 @@ def application(application_id):
     sub_criteria_status_completed = all_status_completed(state)
     form = AssessmentCompleteForm()
     associated_tags = get_associated_tags_for_application(application_id)
+    add_comment_argument = request.args.get("add_comment") == "1"
+    edit_comment_argument = request.args.get("edit_comment") == "1"
+    comment_form = CommentsForm()
+
+    if add_comment_argument and comment_form.validate_on_submit():
+        comment = comment_form.comment.data
+        submit_comment(
+            comment=comment,
+            application_id=application_id,
+            sub_criteria_id="",
+            user_id=g.account_id,
+            theme_id="",
+            comment_type=CommentType.WHOLE_APPLICATION.name,
+        )
+
+        return redirect(
+            url_for(
+                "assessment_bp.application",
+                application_id=application_id,
+                _anchor="comments",
+            )
+        )
+
+    state = get_state_for_tasklist_banner(application_id)
+    flags_list = get_flags(application_id)
+
+    comment_response = get_comments(
+        application_id=application_id,
+        sub_criteria_id="",
+        theme_id="",
+        comment_type=CommentType.WHOLE_APPLICATION.name,
+    )
+
+    # TODO add test for this function in data_operations
+    theme_matched_comments = (
+        match_comment_to_theme(comment_response, "", state.fund_short_name)
+        if comment_response
+        else None
+    )
```

Comment:
Does the `match_comment_to_theme` function just work with an empty second argument I'm guessing?

Might be worth using keyword arguments if we're using a blank string, so it's more clean, something like this maybe:

```suggestion
    # TODO add test for this function in data_operations
    theme_matched_comments = None
    if comment_response:
        theme_matched_comments = match_comment_to_theme(
            comment_response=comment_response,
            themes="",
            fund_short_name=state.fund_short_name
        )
```

After looking at that, it looks like second argument is a list and not a string, should maybe we provide an empty list?  

https://github.com/communitiesuk/funding-service-design-assessment/blob/39cf96a/app/blueprints/services/data_services.py#L669

Reply:
fixed, added keyword arguments